src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -1322,7 +1323,8 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		this.redrawTabLabel(editor, tabIndex, tabContainer, tabLabelWidget, tabLabel);
 
 		// Action
-		const tabAction = isTabSticky ? this.unpinEditorAction : this.closeEditorAction;
+		const hasUnpinAction = isTabSticky && options.tabUnpinButton !== 'off';
+		const tabAction = hasUnpinAction ? this.unpinEditorAction : this.closeEditorAction;
```

Comment:
Here we can decide what action to add to the tabActionBar.

We need to check if the tab is sticky and if unpin action should be visible. If that is the case `tabAction = tabUnpinButton` otherwise `tabAction = tabCloseButton` (this is the fall back).